src/coreclr/jit/lsraarm64.cpp

```diff
@@ -1920,18 +1920,44 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
         }
         else
         {
-            assert((numArgs == 1) || (numArgs == 2) || (numArgs == 3));
+            const bool embHasImmediateOperand = HWIntrinsicInfo::HasImmediateOperand(intrinEmb.id);
+            assert((numArgs == 1) || (numArgs == 2) || (numArgs == 3) || (embHasImmediateOperand && (numArgs == 4)));
 
-            // Special handling for ShiftRightArithmeticForDivide:
+            // Special handling for embedded intrinsics with immediates:
             // We might need an additional register to hold branch targets into the switch table
             // that encodes the immediate
-            if (intrinEmb.id == NI_Sve_ShiftRightArithmeticForDivide)
+            switch (intrinEmb.id)
             {
-                assert(embOp2Node->GetOperandCount() == 2);
-                if (!embOp2Node->Op(2)->isContainedIntOrIImmed())
-                {
-                    buildInternalIntRegisterDefForNode(embOp2Node);
-                }
+                case NI_Sve_ShiftRightArithmeticForDivide:
+                    assert(embHasImmediateOperand);
+                    assert(numArgs == 2);
+                    if (!embOp2Node->Op(2)->isContainedIntOrIImmed())
```

Comment:
nit for future PR - we can have `bool needInternalRegister` that gets set and then after the switch-case just check and do `buildInternalIntRegisterDefForNode(embOp2Node);` if it is true.

Reply:
That sounds better -- I'll include that in my next PR.